File: elemi_programozasi_tetelek_rekurzivan/kereses/kereses.py
def helye1(lista):
    i = 0
    while i<len(lista) and not lista[i]<0:
        i+=1
    return i

def helye2(lista):
    for i, e in enumerate(lista):
        if e<0:
            return i
    return len(lista)

def keres(lista):
    h = helye1(lista)
    return -1 if h==len(lista) else h

def eldontes(lista):
    return helye1(lista)!=len(lista)

lvan = [4, 5, 1, -3, 2]
lnincs = [4, 5, 1, 3, 2]

print(helye1(lvan))
print(helye1(lnincs))
print(helye2(lvan))
print(helye2(lnincs))
print(keres(lvan))
print(keres(lnincs))
print(eldontes(lvan))
print(eldontes(lnincs))

File: elemi_programozasi_tetelek_rekurzivan/kereses/kereses.cpp
#include <iostream>
#include <vector>

using namespace std;

int Helye1(vector<double>& lista)
{
    int i = 0;
    while (i < lista.size() && !(lista[i] < 0))
    {
        i++;
    }
    return i;
}

int Helye2(vector<double>& lista)
{
    for (int i = 0; i < lista.size(); i++)
    {
        if (lista[i] < 0)
        {
            return i;
        }
    }
    return lista.size();
}

int Keres(vector<double>& lista)
{
    int h = Helye1(lista);
    return h == lista.size() ? -1 : h;
}

int Eldontes(vector<double>& lista)
{
    return Helye1(lista) != lista.size();
}

double Elso2(vector<double>& lista)
{
    for (double& elem : lista)
    {
        if (elem < 0)
        {
            return elem;
        }
    }
}

int main()
{
    vector<double> vvan = { 4, 5, 1, -2, 3 };
    vector<double> vnincs = { 4, 5, 1, 2, 3 };

    cout << Helye1(vvan) << endl
        << Helye1(vnincs) << endl
        << Helye2(vvan) << endl
        << Helye2(vnincs) << endl
        << Keres(vvan) << endl
        << Keres(vnincs) << endl
        << Eldontes(vvan) << endl
        << Eldontes(vnincs) << endl;
}
    
